### ISSUE TITLE
Improve cursor handling of rotated TRubberbandLaye

### ISSUE THREAD

**Issue by andersmelander** [graphics32/graphics32#90]
There is an other thing to say when rotating RubberBand  the cursors on Handles are not valid since each cursor depends on the anlge of the Handle to center point ... in my case I disabled this portion of code by detecting FIsDragging value and setting the correct cursor  in my new code before pass to the inherited method: 
TRubberbandLayer.MouseMove
```
  if not FIsDragging then
  begin
    FDragState := GetDragState(X, Y);
    if FDragState = dsMove then
      Screen.Cursor := Cursor
    else
      Screen.Cursor := CURSOR_ID[FDragState];
  end
```
I think it's good adea to add new method
procedure SetStateCursor(ACursor:TCursor):virtual;

and the ne code in TRubberbandLayer.MouseMove
```
  if not FIsDragging then
  begin
    FDragState := GetDragState(X, Y);
    if FDragState = dsMove then
      SetStateCursor(Cursor)
    else
     SetStateCursor( CURSOR_ID[FDragState]);
  end
```

_Originally posted by @lamdalili in https://github.com/graphics32/graphics32/issues/81#issuecomment-499463075_

**Comment by lamdalili** [comment]
Here screenshot soo the center point can be deplaced from center.
![svgedit](https://user-images.githubusercontent.com/40634105/59978106-0f617d80-95d9-11e9-882f-e457cf3d03e3.png)

**Comment by andersmelander** [comment]
I solved this a bit differently. I've added a new virtual method:

    function TRubberbandLayer.GetHandleCursor(DragState: TRBDragState; Angle: integer): TCursor;

and the code in `TRubberbandLayer.MouseMove` now looks like this:

```
if not FIsDragging then
begin
  FDragState := GetDragState(X, Y);
  Angle := DragStateToAngle[FDragState];
  Screen.Cursor := GetHandleCursor(FDragState, Angle);
end
```

The default implementation of `GetHandleCursor` method ignores the value of `DragState` and instead uses the `Angle` (automatically snapped to 45 degrees) to determine the cursor. This means that derived classes can just override the method to manipulate the angle (in the case of a rotated layer) and then call inherited.
